src/services/services.ts

```diff
@@ -1882,41 +1882,38 @@ namespace ts {
             return NavigationBar.getNavigationTree(syntaxTreeCache.getCurrentSourceFile(fileName), cancellationToken);
         }
 
-        function isTsOrTsxFile(fileName: string): boolean {
-            const kind = getScriptKind(fileName, host);
-            return kind === ScriptKind.TS || kind === ScriptKind.TSX;
-        }
-
         function getSemanticClassifications(fileName: string, span: TextSpan): ClassifiedSpan[];
         function getSemanticClassifications(fileName: string, span: TextSpan, format?: SemanticClassificationFormat): ClassifiedSpan[] | ClassifiedSpan2020[] {
-            if (!isTsOrTsxFile(fileName)) {
+            synchronizeHostData();
+            const sourceFile = program && program.getSourceFile(fileName);
+            if (!sourceFile) {

```

Comment:
moves `synchronizeHostData` to the start of the function, _then_ checks for the `program` and `sourceFile`. Only potential behavior change is that `getValidSourceFile` would previously have thrown when there is no source file and now it would return empty results.

Reply:
I think we should still keep getValidSourceFile as calling this on wrong LS is still error...

Reply:
Sure thing, done!